webapp/blog/logic.py

```diff
@@ -16,7 +16,40 @@ def strip_excerpt(raw_html):
     return html.unescape(cleantext).replace("\n", "")
 
 
-def transform_article(article, featured_image=None, author=None):
+def replace_images_with_cloudinary(content):
+    """Prefixes images with cloudinary optimised URLs and adds srcset for
+    image scaling
+
+    :param content: The HTML string to convert
+
+    :returns: Update HTML string with converted images
+    """
+    cloudinary = "https://res.cloudinary.com/"
+
+    urls = [
+        cloudinary + "canonical/image/fetch/q_auto,f_auto,w_650/\g<url>",
+        cloudinary + "canonical/image/fetch/q_auto,f_auto,w_1300/\g<url>",
+        cloudinary + "canonical/image/fetch/q_auto,f_auto,w_1950/\g<url>",
+    ]
+
+    image_match = (
+        r'<img(?P<prefix>[^>]*) src="(?P<url>[^"]+)"(?P<suffix>[^>]*)>'
+    )
+    replacement = (
+        "<img\g<prefix>"
+        f' decoding="async"'
+        f' src="{urls[0]}"'
+        f' srcset="{urls[0]} 650w, {urls[1]} 1300w, {urls[2]} 1950w"'
+        f' sizes="650px"'
```

Comment:
Do we need `1300w` or `1950w` is we have only `650px` in sizes?

This seems to be only for blog content, which we always have at 650px width, so we will never need bigger images. If anything, we could use smaller images on smaller screens.

Reply:
1300w and 1950w are for retina+ displays

I've added one smaller size for below 400px. The way the columns resize means that at ~400px the columns go 100% width so I don't think it's worth having a complicated srcset. wdyt?

Reply:
Ah, OK - I was debugging it incorrectly. Inspecting HTML Chrome was telling me 650px image is used (as this is img src), but looking at network panel I see that bigger is downloaded for retina.

Thanks for adding one smaller one. Works well.